services/src/main/scala/cromwell/services/healthmonitor/impl/CommonMonitoredSubsystems.scala

```diff
@@ -0,0 +1,49 @@
+package cromwell.services.healthmonitor.impl
+
+import cromwell.core.docker.{DockerCliClient, DockerCliKey}
+import cromwell.services.EngineServicesStore
+import cromwell.services.healthmonitor.HealthMonitorServiceActor.{MonitoredSubsystem, OkStatus, SubsystemStatus}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CommonMonitoredSubsystems {
+  implicit val ec: ExecutionContext
+
+  lazy val DockerHub = MonitoredSubsystem("DockerHub", checkDockerhub _)
+  lazy val EngineDb = MonitoredSubsystem("Engine Database", checkEngineDb _)
+
+  /**
+    * Demonstrates connectivity to DockerHub by periodically pulling a small image. Remove the image afterwards in
+    * order to ensure that subsequent attempts aren't showing a false positive
+    */
+  private def checkDockerhub(): Future[SubsystemStatus] = {
+    println("DOCKER")
+ //   Future {
+      // Using a tiny docker image the user is very unlikely to be actively using
+      println("YO")
+      val alpine = DockerCliKey("alpine", "latest") // FIXME: make a clone of alpine in our dockerhub and use that
+      println("HO")
+
+    val res = DockerCliClient.pull(alpine) flatMap { _ => DockerCliClient.rmi(alpine) }
+
+//      val res = for {
+//        p <- DockerCliClient.pull(alpine)
+//        r <- DockerCliClient.rmi(alpine)
+//      } yield r
+
+      println("FOO: " + res)
+
+//      res.get
+ //   } map { _ => OkStatus }
```

Comment:
got some `println`s and commented code around here 😄

Reply:
You remember that comment about how i think i found a bug due to my docker daemon being borked? apparently i managed to leave some of my "wtf" code in after one of my checkpoint pushes ;)